syncd/syncd_reinit.cpp

```diff
@@ -0,0 +1,558 @@
+
+        std::string strLane;
+        std::string strPortId;
+
+        sai_serialize_primitive(lane, strLane);
+        sai_serialize_primitive(portId, strPortId);
+
+        // TODO use multi or hmset
+        g_redisClient->hset(LANES, strLane, strPortId);
+    }
+}
+
+std::vector<std::string> redisGetAsicStateKeys()
+{
+    SWSS_LOG_ENTER();
+
+    return g_redisClient->keys("ASIC_STATE:*");
+}
+
+void redisClearVidToRidMap()
+{
+    SWSS_LOG_ENTER();
+
+    g_redisClient->del(VIDTORID);
+}
+
+void redisClearRidToVidMap()
+{
+    SWSS_LOG_ENTER();
+
+    g_redisClient->del(RIDTOVID);
+}
+
+std::unordered_map<sai_object_id_t, sai_object_id_t> redisGetObjectMap(std::string key)
+{
+    SWSS_LOG_ENTER();
+
+    auto hash = g_redisClient->hgetall(key);
+
+    std::unordered_map<sai_object_id_t, sai_object_id_t> map;
+
+    for (auto &kv: hash)
+    {
+        const std::string &str_key = kv.first;
+        const std::string &str_value = kv.second;
+
+        sai_object_id_t objectIdKey;
+        sai_object_id_t objectIdValue;
+
+        int index = 0;
+
+        sai_deserialize_primitive(str_key, index, objectIdKey);
+
+        index = 0;
+
+        sai_deserialize_primitive(str_value, index, objectIdValue);
+
+        map[objectIdKey] = objectIdValue;
+    }
+
+    return map;
+}
+
+std::unordered_map<sai_object_id_t, sai_object_id_t> redisGetVidToRidMap()
+{
+    SWSS_LOG_ENTER();
+
+    return redisGetObjectMap(VIDTORID);
+}
+
+std::unordered_map<sai_object_id_t, sai_object_id_t> redisGetRidToVidMap()
+{
+    SWSS_LOG_ENTER();
+
+    return redisGetObjectMap(RIDTOVID);
+}
+
+void helperCheckLaneMap()
+{
+    SWSS_LOG_ENTER();
+
+    auto redisLaneMap = redisGetLaneMap();
+
+    auto laneMap = saiGetHardwareLaneMap();
+
+    if (redisLaneMap.size() == 0)
+    {
+        SWSS_LOG_INFO("no lanes defined in redis, seems like it is first syncd start");
+
+        // TODO put ports to db ?
+        redisSaveLaneMap(laneMap);
+
+        redisLaneMap = laneMap; // copy
+    }
+
+    if (laneMap.size() != redisLaneMap.size())
+    {
+        SWSS_LOG_ERROR("lanes map size differ: %lu vs %lu", laneMap.size(), redisLaneMap.size());
+
+        exit(EXIT_FAILURE);
+    }
+
+    for (auto kv: laneMap)
+    {
+        sai_uint32_t lane = kv.first;
+        sai_object_id_t portId = kv.second;
+
+        if (redisLaneMap.find(lane) == redisLaneMap.end())
+        {
+            SWSS_LOG_ERROR("lane %u not found in redis", lane);
+
+            exit(EXIT_FAILURE);
+        }
+
+        if (redisLaneMap[lane] != portId)
+        {
+            // if this happens, we need to remap VIDTORID and RIDTOVID
+            SWSS_LOG_ERROR("FIXME: lane port id differs: %llx vs %llx, port ids must be remapped", portId, redisLaneMap[lane]);
+
+            exit(EXIT_FAILURE);
+        }
+    }
+}
+
+sai_object_id_t redisGetDefaultVirtualRouterId()
+{
+    SWSS_LOG_ENTER();
+
+    auto redisVrId = g_redisClient->hget(HIDDEN, DEFAULT_VIRTUAL_ROUTER_ID);
```

Comment:
yes, thats the point, that those objcts are created inside sai implementation and are "hidden"